components/infobox/wikis/halo/infobox_league_custom.lua

```diff
@@ -17,59 +18,55 @@ local String = require('Module:StringUtils')
 local Variables = require('Module:Variables')
 
 local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
-local League = Lua.import('Module:Infobox/League', {requireDevIfEnabled = true})
+local League = Lua.import('Module:Infobox/League/temp', {requireDevIfEnabled = true})
 
 local Widgets = require('Module:Infobox/Widget/All')
 local Cell = Widgets.Cell
 local Title = Widgets.Title
 local Center = Widgets.Center
 
-local CustomLeague = Class.new()
+---@class HaloLeagueInfobox: InfoboxLeagueTemp
+local CustomLeague = Class.new(League)
 local CustomInjector = Class.new(Injector)
 
-local _args
-
+---@param frame Frame
+---@return Html
 function CustomLeague.run(frame)
-	local league = League(frame)
-	_args = league.args
-
-	_args.liquipediatiertype = _args.liquipediatiertype or _args.tiertype
+	local league = CustomLeague(frame)
+	league:setWidgetInjector(CustomInjector(league))
 
-	league.addToLpdb = CustomLeague.addToLpdb
-	league.createWidgetInjector = CustomLeague.createWidgetInjector
-	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
-	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+	league.args.liquipediatiertype = league.args.liquipediatiertype or league.args.tiertype
 
 	return league:createInfobox()
 end
 
-function CustomLeague:createWidgetInjector()
-	return CustomInjector()
-end
-
-function CustomInjector:addCustomCells(widgets)
-	return {
-		Cell{name = 'Number of teams', content = {_args.team_number}},
-		Cell{name = 'Number of players', content = {_args.player_number}},
-	}
-end
-
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
 function CustomInjector:parse(id, widgets)
-	if id == 'gamesettings' then
+	local args = self.caller.args
+
+	if id == 'custom' then
 		return {
-			Cell{name = 'Game version', content = {Game.name{game = _args.game}}},
+			Cell{name = 'Number of teams', content = {args.team_number}},
+			Cell{name = 'Number of players', content = {args.player_number}},
 		}
-	elseif id == 'customcontent' then
-		--maps
-		if not String.isEmpty(_args.map1) then
-			table.insert(widgets, Title{name = 'Maps'})
-			table.insert(widgets, Center{content = CustomLeague:_makeMapList()})
-		end
+	elseif id == 'gamesettings' then
+		return {
+			Cell{name = 'Game version', content = {Game.name{game = args.game}}},
```

Comment:
This is causing the Game cell to take up 2 lines. However that is a Module:Info issue (similar to #3726 ) rather than an issue here

Reply:
don't think it is due to this PR, think it is #3655 causing it
could you check if it actually changes between dev and non dev?

Reply:
Halo does not have this issue. This meant to be for Rainbow Six (i apparently cant read what file im in)

Reply:
same applies for r6 though

Reply:
Ill make an adjustment to the R6 Info module to account for this